### PR TITLE
CI - parallelize everything that can be paralellized

### PR DESCRIPTION
**Pull request by lwitkowski**
Static checks, unit and integrationt tests extracted to separate workflows, no test results publishing

Why?
- all checks completed under 1:30
- all checks listed nicely in PR: `10 successful checks`
- less spam on mailbox (test reports)

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        maven_profiles: [
+            "jackson-classic", "jsonb-classic",
+            "jackson-reactive", "jsonb-reactive",
+            "jackson-classic,quarkus-1.4", "jsonb-classic,quarkus-1.4",
+            "jackson-reactive,quarkus-1.11", "jsonb-reactive,quarkus-1.11"
```
Comment (pazkooda) [inline .github/workflows/integration-tests.yaml:20]: Why `11` and not latest  (`13`)?

File: .github/workflows/maven-regression.yaml [hunk 2]
```diff
@@ -5,12 +5,13 @@ on:
...
   workflow_dispatch:
 
+
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        version: [ "1.4", "1.5", "1.6", "1.7", "1.8", "1.9", "1.10", "1.11", "1.12" ]
+        version: [ "1.5", "1.6", "1.7", "1.8", "1.9", "1.10", "1.11", "1.12" ]
```
Comment (pazkooda) [inline .github/workflows/maven-regression.yaml:14]: Why `1.4` is gone?

Comment (lwitkowski) [reply .github/workflows/maven-regression.yaml:14]: It duplicates integration-tests.yaml workflow, same with latest 1.13 (default, if no profile specified)

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (lwitkowski) [reply .github/workflows/integration-tests.yaml:20]: 1.11 is the oldest quarkus for which we support reactive version, I moved it from 1.12 few days ago when I run the tests in 1.11 and it worked :)

Comment (lwitkowski) [reply .github/workflows/integration-tests.yaml:20]: Regarding 1.13: it is used when no `quarkus-XX` profile is specified

**Review by pazkooda — APPROVED** [review]

**Pull request merged**